feat(managers): implement algorithm for peer discovery in SessionManager (#128)
witnet/witnet-rust: docs/architecture/managers/sessions-manager.md

These are the messages supported by the sessions manager handlers:

| Message       | Input type                                | Output type           | Description                                    |
|---------------|-------------------------------------------|-----------------------|------------------------------------------------|
| Register      | `SocketAddr, Addr<Session>, SessionType`  | `SessionsResult<()>`  | Request to register a new session              |
| Unregister    | `SocketAddr, SessionType`                 | `SessionsResult<()>`  | Request to unregister a session                |
| Update        | `SocketAddr, SessionType, SessionStatus`  | `SessionsResult<()>`  | Request to update the status of a session      |

> | Anycast<T>    | `T`                                       | `()`                  | Request to send a T message to a random Session|

Reviewer: Please put all messages between backticks (``)

    })
    .wait(ctx);
```
#### Anycast<T>

> The handling of this messages is basically just calling the method `get_random_anycast_session` from the

Reviewer: ```suggestion
The handler for `Anycast<T>` messages is basically just calling the method `get_random_anycast_session` from the
```

> [`Sessions`][sessions] library to obtain a random Session and send a `T` message.

Reviewer: ```suggestion
[`Sessions`][sessions] library to obtain a random `Session` and forward the `T` message to it.
```

Reviewer: Shall we mention here what happens to the return value of the delegated call?

Author: Added rust code with the function where it happens

### Outgoing messages: Connections Manager -> Others

These are the messages sent by the sessions manager:

| Message               | Destination           | Input type    | Output type                       | Description                                           |
|-----------------------|-----------------------|---------------|-----------------------------------|-------------------------------------------------------|
| GetServerAddress      | ConfigManager         | `()`          | `Option<SocketAddr>`              | Request the config server address                     |
| GetConnLimits         | ConfigManager         | `()`          | `Option<(u16, u16)>`              | Request the config connections limits                 |

> | GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request the address of a peer                         |

Reviewer: ```suggestion
| GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request a random peer address                        |
```

| OutboundTcpConnect    | ConnectionsManager    | `SocketAddr`  | `()`                              | Request a TCP conn to an address                      | 

> | Anycast<GetPeers>     | SessionsManager       | `()`          | `()`                              | Request to send a GetPeers message to a random Session|

Reviewer: Please add backticks (``) around the message names and destinations

> | Anycast<GetPeers>     | SessionsManager       | `()`          | `()`                              | Request to send a GetPeers message to a random Session|

Reviewer: ```suggestion
| `Anycast<GetPeers>`     | `SessionsManager`       | `()`          | `()`                              | Request to forward a GetPeers message to one randomly selected `Session`|
```

For further information, see [`ConnectionsManager`][connections_manager].

#### Anycast<GetPeers>

> This message is periodically sent to itself ([`SessionsManager`][sessions_manager]).

Reviewer: What for is this sent to itself? Please explain.

Author: Done!

It is a best effort message, its return value is not processed and the [`SessionsManager`][sessions_manager] 
actor does not even wait for its response after sending it.

> This message request to send a `GetPeers` message to a random Session. 

Reviewer: ```suggestion
This message causes `SessionManager` to forward a `GetPeers` message to one randomly selected `Session` actor. 
```
